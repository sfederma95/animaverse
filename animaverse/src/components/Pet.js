import React from 'react';

function Pet({src,type,level}){
    return(
        <div>
            <img alt={type} src={src}/>
            <p>Type: {type}</p>
            <p>Level: {level}</p>
            <button>Select</button>
        </div>
    )
}

export default Pet;