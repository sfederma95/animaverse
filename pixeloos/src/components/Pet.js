import React from 'react';

function Pet({src,type,level, buttonFn, n}){
    return(
        <div className='adopt-pets'>
            <p>{n} / 6</p>
            <img onClick={buttonFn} alt={type} src={src}/>
            <p>Type: {type}</p>
            <p>Level: {level}</p>
        </div>
    )
}

export default Pet;