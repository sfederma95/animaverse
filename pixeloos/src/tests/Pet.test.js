import React from 'react';
import {render} from '@testing-library/react';
import {MemoryRouter} from 'react-router';
import Pet from '../components/Pet';
import {UserProvider} from './testSetup';

it ('renders pet component without crashing', function(){
    render(
        <MemoryRouter>
            <UserProvider>
                <Pet type='dog' level={1} n={1}/>
            </UserProvider>
        </MemoryRouter>
    )
})

it ('matches snapshot', function(){
    const {asFragment} = render (
        <MemoryRouter>
            <UserProvider>
                <Pet type='dog' level={1} n={1}/>
            </UserProvider>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot();
})