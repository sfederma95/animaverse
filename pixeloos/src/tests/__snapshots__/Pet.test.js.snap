// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="adopt-pets"
  >
    <p>
      1 / 6
    </p>
    <img
      alt="dog"
    />
    <p>
      Type: dog
    </p>
    <p>
      Level: 1
    </p>
  </div>
</DocumentFragment>
`;
